runtime_lib/controller/main.cpp

```diff
@@ -820,6 +822,60 @@ void xaie_l2_dma_init(int col) {
   xaie::out32(xaie::getTileAddr(col, 0) + 0x00033008, 0xFF);
 }
 
+#ifdef ARM_CONTROLLER
+
+// Defining the NPI base and registers we use to reset the array
+constexpr uint64_t npi_base = 0xF70A0000UL;
+constexpr auto NPI_MASK_REG = 0x0;
+constexpr auto NPI_VAL_REG = 0x4;
+constexpr auto NPI_LOCK_REG = 0xC;
+void xaie_array_reset() {
+
+  // Getting a pointer to NPI
+  auto *npib = (volatile uint32_t *)(npi_base);
```

Comment:
```suggestion
  auto npib = (volatile uint32_t *)(npi_base);
```
In that case it is obvious it is a pointer, but if you want you can also keep the `*` to insist on it.